src/NimBLEDevice.cpp

```diff
@@ -479,15 +479,27 @@ bool NimBLEDevice::setPower(int8_t dbm) {
 
 /**
  * @brief Get the transmission power.
- * @return The power level currently used in dbm.
- * @note ESP32S3 only returns 0xFF as of IDF 5, so this function will return 20dbm.
+ * @return The power level currently used in dbm or 0xFF on error.
  */
 int NimBLEDevice::getPower() {
 # ifdef ESP_PLATFORM
 #  ifdef CONFIG_IDF_TARGET_ESP32P4
     return 0xFF; // CONFIG_IDF_TARGET_ESP32P4 does not support esp_ble_tx_power_get
 #  else
     int pwr = esp_ble_tx_power_get(ESP_BLE_PWR_TYPE_DEFAULT);
+
+#   if defined(CONFIG_IDF_TARGET_ESP32C3) || defined(CONFIG_IDF_TARGET_ESP32S3)
+    // workaround for bug when "enhanced tx power" was added
+    if (pwr == 0xFF) {
+        pwr = esp_ble_tx_power_get(ESP_BLE_PWR_TYPE_CONN_HDL3);
```

Comment:
Should this be moved into `getPowerLevel`, and have this function call it?

Reply:
Hmm, yes it probably should 👍

Reply:
done